include/seqan3/core/tuple_utility.hpp

```diff
@@ -0,0 +1,184 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides utility functions for tuple like interfaces.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <meta/meta.hpp>
+
+#include <utility>
+
+#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/core/type_list.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Helper function for seqan3::tuple_split.
+ * \ingroup core
+ *
+ * \tparam    beg A template value containing the start position from where to extract the values.
+ *
+ * \param[in] t   The original tuple to split.
+ * \param[in] idx An std::index_sequence with all indices that should be extracted beginning at `beg`.
+ *
+ * \returns A new tuple with the extracted elements.
+ */
+template <size_t beg,
+          template <typename ...> typename tuple_t,
+          size_t ... Is,
+          typename ...ts>
+          requires tuple_like_concept<tuple_t<ts...>> && tuple_like_concept<tuple_t<>>
+constexpr auto tuple_split(tuple_t<ts...> const & t, std::index_sequence<Is...> const & SEQAN3_DOXYGEN_ONLY(idx))
+{
+    return tuple_t<std::tuple_element_t<beg + Is, tuple_t<ts...>>...>{std::get<beg + Is>(t)...};
+}
+
+//!\copydoc seqan3::detail::tuple_split
+template <size_t beg,
+          template <typename ...> typename tuple_t,
+          size_t ... Is,
+          typename ...ts>
+          requires tuple_like_concept<tuple_t<ts...>> && tuple_like_concept<tuple_t<>>
+constexpr auto tuple_split(tuple_t<ts...> && t, std::index_sequence<Is...> const & SEQAN3_DOXYGEN_ONLY(idx))
+{
+    return tuple_t<std::tuple_element_t<beg + Is, tuple_t<ts...>>...>{std::move(std::get<beg + Is>(t))...};
+}
+}
+
+namespace seqan3
+{
+
+/*!\brief Splits a tuple like data structure at the given position.
+ * \ingroup core
+ *
+ * \tparam    pivot_c A template value specifying the split position.
+ * \param[in] t       The original tuple to split.
+ *
+ * \returns A new tuple of tuples with the left side of the split and the right side of the split.
```

Comment:
basically it returns `[t_0, ..., t_{p-1}], [t_p, ..., t_{n-1}]`;

corner cases:
 * `p=0`: `[], [t_0, ..., t_{n-1}]`
 * `p=n`: `[t_0, ..., t_{n-1}], []`

can you make this more explicit in the docs?